code/infrastructure-api/src/config/routes.js

```diff
@@ -1,18 +1,32 @@
 import status from '../controllers/status';
 import stack from '../controllers/stackController';
+import stacks from '../controllers/stacksController';
 import volume from '../controllers/volumeController';
 import verifyToken from '../auth/authMiddleware';
 import permissionWrapper from '../auth/permissionMiddleware';
 
+const STACKS_LIST = 'stacks:list';
+const STACKS_OPEN = 'stacks:open';
+const STACKS_CREATE = 'stacks:create';
+const STACKS_DELETE = 'stacks:delete';
+const STORAGE_LIST = 'storage:list';
+const STORAGE_CREATE = 'storage:create';
+const STORAGE_DELETE = 'storage:delete';
+
 function configureRoutes(app) {
   app.get('/status', status.status);
   app.all('*', verifyToken); // Routes above this line are not auth checked
-  app.post('/stacks', permissionWrapper('stacks:create'), stack.createStackValidator, stack.createStack);
-  app.delete('/stacks', permissionWrapper('stacks:delete'), stack.coreStackValidator, stack.deleteStack);
-  app.get('/volumes', permissionWrapper('storage:list'), volume.listVolumes);
-  app.post('/volumes', permissionWrapper('storage:create'), volume.createVolumeValidator, volume.createVolume);
-  app.delete('/volumes', permissionWrapper('storage:delete'), volume.coreVolumeValidator, volume.deleteVolume);
-  app.post('/volumes/query', permissionWrapper('storage:list'), volume.coreVolumeValidator, volume.queryVolume);
+  app.get('/stacks', permissionWrapper(STACKS_LIST), stacks.listStacks);
```

Comment:
Now might be the time to consider routers - https://expressjs.com/en/guide/routing.html - express.Router
Though maybe lets get this merged and refactor if you agree.